CRFModuleValidation/MasterScheduledActivityManager.swift

```diff
@@ -0,0 +1,563 @@
+//
+//  MasterScheduledActivityManager.swift
+//  JourneyPRO
+//
+//  Copyright © 2017 Sage Bionetworks. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification,
+// are permitted provided that the following conditions are met:
+//
+// 1.  Redistributions of source code must retain the above copyright notice, this
+// list of conditions and the following disclaimer.
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+import BridgeSDK
+import BridgeAppSDK
+import ResearchUXFactory
+import UserNotifications
+import ResearchSuiteUI
+
+enum ScheduleLoadState {
+    case firstLoad
+    case cachedLoad
+    case fromServer
+}
+
+class MasterScheduledActivityManager: ScheduledActivityManager {
+    
+    static let shared = MasterScheduledActivityManager(delegate: nil)
+    
+    // Save a pointer to today's activities in case the paged group does not include them
+    var scheduleSections: [ScheduleSection] = []
+    var schedules: [Schedule] = []
+    
+    var clinicDay0Schedule: SBBScheduledActivity?
+    
+    let scheduleUpdatedNotificationName = Notification.Name("MasterScheduledActivityManager.scheduleUpdated")
+    
+    let studyDuration: DateComponents = {
+        var studyDuration = DateComponents()
+        studyDuration.day = 14
+        return studyDuration
+    }()
+    
+    var enrollment: Date = {
+        return (UIApplication.shared.delegate as! AppDelegate).currentUser.createdOn.startOfDay()
+    }()
+    
+    var startStudy: Date {
+        return Calendar.gregorian.startOfDay(for: dayOne ?? enrollment)
```

Comment:
startStudy date should roll off of day one only. Since using externalIDs they will have a creation date that isn't an accurate enrollment date.

Reply:
As far as I can tell, that's actually just a placeholder until the first activity/survey gets completed. The start-of-study date doesn't appear in the UI until then either, so the participant never actually sees the "enrollment" date being claimed as the start of the study.